sources/networkmanager/vpc-attachment.go

```diff
@@ -0,0 +1,66 @@
+package networkmanager
+
+import (
+	"context"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/networkmanager"
+	"github.com/aws/aws-sdk-go-v2/service/networkmanager/types"
+	"github.com/overmindtech/aws-source/sources"
+	"github.com/overmindtech/sdp-go"
+)
+
+func vpcAttachmentGetFunc(ctx context.Context, client *networkmanager.Client, scope, query string) (*types.VpcAttachment, error) {
+	out, err := client.GetVpcAttachment(ctx, &networkmanager.GetVpcAttachmentInput{
+		AttachmentId: &query,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return out.VpcAttachment, nil
+}
+
+// TODO: connect core-network here
+func vpcAttachmentItemMapper(scope string, awsItem *types.VpcAttachment) (*sdp.Item, error) {
+	attributes, err := sources.ToAttributesCase(awsItem)
+
+	if err != nil {
+		return nil, err
+	}
+
+	item := sdp.Item{
+		Type:            "networkmanager-vpc-attachment",
+		UniqueAttribute: "Attachment.AttachmentId",
+		Attributes:      attributes,
+		Scope:           scope,
+	}
+
+	return &item, nil
```

Comment:
I think we're missing some links here. The attachment has references to core networks, subnets etc that we can link. For something like a core network, we can create the links here even though we haven't created the source yet because we know what the source will be called based on the [naming convention](https://github.com/overmindtech/aws-source?tab=readme-ov-file#naming-conventions) and [cli arguments](https://docs.aws.amazon.com/cli/latest/reference/networkmanager/list-core-networks.html) so it'd be `networkmanager-core-network`